zebedee/client.go

```diff
@@ -389,6 +391,46 @@ func (c *Client) GetBulletin(ctx context.Context, userAccessToken, lang, uri str
 		return bulletin, err
 	}
 
+	if !bulletin.Description.LatestRelease {
+		// Resolve the latest release URI
+		latestUrl := fmt.Sprintf("%s/latest", bulletin.URI[:strings.LastIndex(bulletin.URI, "/")])
+		t, err := c.GetPageTitle(ctx, userAccessToken, collectionID, lang, latestUrl)
+		if err != nil {
+			log.Error(ctx, "error finding latest release URI", err, log.Data{"url": latestUrl})
+			bulletin.LatestReleaseURI = latestUrl
+		} else {
+			bulletin.LatestReleaseURI = t.URI
+		}
+	}
+
+	related := [][]Link{
+		bulletin.RelatedBulletins,
+		bulletin.Links,
+	}
+
+	// Concurrently resolve any URIs where we need more data from another page
+	var wg sync.WaitGroup
+	sem := make(chan int, 10)
```

Comment:
I assume the use of sem channel is to limit the number of concurrent requests to Bulletins API as to not DOS the service?